ArcGISRuntimeSDKQt_QMLSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.qml

```diff
@@ -0,0 +1,359 @@
+            
+            ViewpointExtent {
+                Envelope {
+                    xMin: -9813547.35557238
+                    yMin: 5129980.36635111
+                    xMax: -9813185.0602376
+                    yMax: 5130215.41254146
+                    spatialReference: SpatialReference.createWebMercator();
+                }
+            }
+
+            // Add the layer with electric distribution lines.
+            FeatureLayer {
+                id: lineLayer
+
+                ServiceFeatureTable {
+                    url: featureLayerUrl + "/115"
+                }
+
+                renderer: SimpleRenderer {
+                    symbol: featureLineSymbol
+                }
+
+                onSelectFeaturesStatusChanged: checkSelectionStatus();
+            }
+
+            // Add the layer with electric devices.
+            FeatureLayer {
+                id: deviceLayer
+
+                ServiceFeatureTable {
+                    url: featureLayerUrl + "/100"
+                }
+
+                onSelectFeaturesStatusChanged: checkSelectionStatus();
+            }
+        }
+
+        GraphicsOverlay {
+            id: unGraphicsOverlay
+        }
+
+        onMouseClicked: {
+            const screenX = mouse.x;
+            const screenY = mouse.y;
+            const tolerance = 10;
+            const returnPopups = false;
+            clickPoint = mouse.mapPoint;
+            mapView.identifyLayers(screenX, screenY, tolerance, returnPopups);
+        }
+
+        onIdentifyLayersStatusChanged : {
+            if (identifyLayersStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            const results = mapView.identifyLayersResults;
+            const result = results[0];
+            if ( !result ) {
+                dialogText.text = qsTr("Could not identify location.")
+                traceCompletedDialog.open();
+                return;
+            }
+
+            identifiedFeature = result.geoElements[0];
+            utilityNetworkSource = utilityNetwork.definition.networkSource(identifiedFeature.featureTable.tableName);
+
+            if (utilityNetworkSource.sourceType === Enums.UtilityNetworkSourceTypeJunction) {
+                const assetGroupFieldName = identifiedFeature.featureTable.subtypeField;
+                const assetGroupCode = identifiedFeature.attributes.attributeValue(assetGroupFieldName);
+
+                let assetGroup;
+                for (let i = 0; i < utilityNetworkSource.assetGroups.length; i++) {
+                    if (utilityNetworkSource.assetGroups[i].code === assetGroupCode) {
+                        assetGroup = utilityNetworkSource.assetGroups[i];
+                        break;
+                    }
+                }
+
+                const assetTypeCode = identifiedFeature.attributes.attributeValue("assettype");
+
+                let assetType;
+                for (let j = 0; j < assetGroup.assetTypes.length; j++) {
```

Comment:
with `let`, i is now block-scoped within the for-loop, so you can use `i` again in this loop. Optional, but using i, j, k, etc.. always looks like nested loops to me - or quirky Qml scoping rules.